nautobot/core/jobs/__init__.py

```diff
@@ -209,7 +216,45 @@ class Meta:
         soft_time_limit = 1800
         time_limit = 2000
 
-    def run(self, *, content_type, csv_data=None, csv_file=None):
+    def _perform_atomic_operation(self, data, serializer_class, queryset):
+        new_objs = []
+        with contextlib.suppress(AbortTransaction):
+            with transaction.atomic():
+                new_objs, validation_failed = self._perform_operation(data, serializer_class, queryset)
+                if validation_failed:
+                    raise AbortTransaction
+                return new_objs, validation_failed
+        # If validation failed return an empty list, since all objs created where rolled back
+        self.logger.error("Rolling back all %s records.", len(new_objs))
```

Comment:
This message isn't itself an error, perhaps `logger.info` would be more appropriate?